pkgs/test/nixos-functions/default.nix

```diff
@@ -0,0 +1,22 @@
+{ pkgs, lib, stdenv, ... }:
+
+lib.optionalAttrs stdenv.hostPlatform.isLinux (
+  pkgs.recurseIntoAttrs {
+
+    nixos-test = (pkgs.nixos {
+      boot.loader.grub.enable = false;
+      fileSystems."/".device = "/dev/null";
+    }).toplevel;
+
+    nixosTest-test = pkgs.nixosTest ({ lib, pkgs, ... }: {
+      name = "nixosTest-test";
+      machine = { pkgs, ... }: {
+        environment.systemPackages = [ pkgs.hello ];
+      };
+      testScript = ''
+        $machine->succeed("hello");
+      '';
+    });
+
+  }
+)
```

Comment:
What’s this file for?

Reply:
This file is a test that makes sure that the `nixos` and `nixosTest` functions work. It's far from a perfect test suite, but better than not checking them at all on hydra. To test: `nix-build -A tests.nixos-functions`

Reply:
I’d add a comment at the top to clarify.

Reply:
Done.